app/controllers/my_modules_controller.rb

```diff
@@ -249,109 +249,24 @@ def results
 
   def archive
     @archived_results = @my_module.archived_results
-    current_team_switch(@my_module
-                                .experiment
-                                .project
-                                .team)
+    current_team_switch(@my_module.experiment.project.team)
   end
 
-  # Complete/uncomplete task
-  def toggle_task_state
-    respond_to do |format|
-      if can_complete_module?(@my_module)
-        @my_module.completed? ? @my_module.uncomplete : @my_module.complete
-        completed = @my_module.completed?
-        if @my_module.save
-          task_completion_activity
-
-          # Render new button HTML
-          if completed
-            new_btn_partial = 'my_modules/state_button_uncomplete.html.erb'
-          else
-            new_btn_partial = 'my_modules/state_button_complete.html.erb'
-          end
+  def update_state
+    # ### ToDo
+    # Add logic for updating state here
+    # Check if you can move to that state
+    # Permit params
 
-          format.json do
-            render json: {
-              new_btn: render_to_string(partial: new_btn_partial),
-              completed: completed,
-              module_header_due_date: render_to_string(
-                partial: 'my_modules/module_header_due_date.html.erb',
-                locals: { my_module: @my_module }
-              ),
-              module_state_label: render_to_string(
-                partial: 'my_modules/module_state_label.html.erb',
```

Comment:
Do we still need partial `my_modules/module_state_label.html.erb`? Probably you can remove it.